src/CodeInfoTools.jl

```diff
@@ -558,7 +558,11 @@ function code_inferred(mi::Core.Compiler.MethodInstance;
     src.inferred && return src
     Core.Compiler.validate_code_in_debug_mode(result.linfo, 
                                               src, "lowered")
-    frame = Core.Compiler.InferenceState(result, src, false, interp)
+    frame = @static if VERSION.major == 1 && VERSION.minor < 8
```

Comment:
Maybe use
```suggestion
    frame = @static if VERSION < v"1.8"
```
(BTW I just learnt today that usually `@static` is not needed). Maybe it would be even better to figure out the Github commit in the Julia base repo where this change was introduced to be able to branch more accurately.